File: frontend/src/types/strategy.ts
export type StrategyConfidenceLevel =
  | "insufficient_data"
  | "weak"
  | "moderate"
  | "high";

export interface StrategyConfidence {
  confidence: StrategyConfidenceLevel;
  basis: string;
  sampleSize: number;
  sourceInputs: Record<string, unknown>;
  knownUncertainty: string[];
  insufficientData: string[];
  userOverrides: Record<string, unknown> | null;
}

export interface StrategySignal {
  id: string;
  category: string;
  label: string;
  message: string;
  basis: string;
  severity: "info" | "caution" | "positive";
  confidence: StrategyConfidence;
  sourceInputs: Record<string, unknown>;
}

export interface StrategyActionCandidate {
  id: string;
  category: string;
  title: string;
  rationale: string;
  basis: string;
  confidence: StrategyConfidence;
  sourceInputs: Record<string, unknown>;
  advisoryOnly: true;
}

export interface StrategyInsufficientDataItem {
  reason: string;
  message: string;
  sourceInputs: Record<string, unknown>;
}

export interface StrategySampleSize {
  opportunities: number;
  applications: number;
  submittedApplications: number;
  responses: number;
  compassEvaluations: number;
  artifactPerformanceRecords: number;
}

export interface SearchTrackStrategySummary {
  contractVersion: "careero.contracts.v1";
  workspaceId: string;
  workspaceName: string;
  generatedAt: string;
  summary: string;
  basis: string;
  confidence: StrategyConfidence;
  sampleSize: StrategySampleSize;
  sourceInputs: Record<string, unknown>;
  knownUncertainty: string[];
  insufficientData: StrategyInsufficientDataItem[];
  signals: StrategySignal[];
  compensationAlignment: {
    summary: string;
    basis: string;
    confidence: StrategyConfidence;
    observations: Array<Record<string, unknown>>;
  };
  skillGapThemes: StrategySignal[];
  roleMarketPositioning: {
    summary: string;
    basis: string;
    confidence: StrategyConfidence;
    themes: string[];
  };
  careerNarrativeThemes: StrategySignal[];
  retrospective: {
    summary: string;
    basis: string;
    confidence: StrategyConfidence;
    notes: string[];
  };
  actionCandidates: StrategyActionCandidate[];
  warnings: string[];
}

export interface CrossTrackStrategyComparison {
  generatedAt: string;
  basis: string;
  confidence: StrategyConfidence;
  tracks: Array<{
    workspaceId: string;
    workspaceName: string;
    summary: string;
    sampleSize: Record<string, unknown>;
    signalCount: number;
    warningCount: number;
  }>;
  signals: StrategySignal[];
  insufficientData: StrategyInsufficientDataItem[];
  warnings: string[];
}

export interface CareerStrategySummary {
  contractVersion: "careero.contracts.v1";
  generatedAt: string;
  summary: string;
  workspaceId: string | null;
  workspaceName: string | null;
  activeTrack: SearchTrackStrategySummary | null;
  tracks: SearchTrackStrategySummary[];
  crossTrackComparison: CrossTrackStrategyComparison | null;
  warnings: string[];
}

File: backend/app/schemas/strategy.py
from __future__ import annotations

import uuid
from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, ConfigDict, Field


def to_camel(value: str) -> str:
    first, *rest = value.split("_")
    return first + "".join(part.capitalize() for part in rest)


class StrategyModel(BaseModel):
    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True,
    )


StrategyInsufficientDataReason = Literal[
    "empty_workspace",
    "few_opportunities",
    "few_applications",
    "few_outcomes",
    "missing_compass_evaluations",
    "missing_compensation_ranges",
    "missing_artifact_performance",
    "missing_source_history",
    "stale_track",
    "unknown",
]


class StrategyConfidence(StrategyModel):
    confidence: Literal["insufficient_data", "weak", "moderate", "high"]
    basis: str
    sample_size: int
    source_inputs: dict[str, Any] = Field(default_factory=dict)
    known_uncertainty: list[str] = Field(default_factory=list)
    insufficient_data: list[StrategyInsufficientDataReason] = Field(
        default_factory=list
    )
    user_overrides: dict[str, Any] | None = None


class StrategyInsufficientDataItem(StrategyModel):
    reason: StrategyInsufficientDataReason
    message: str
    source_inputs: dict[str, Any] = Field(default_factory=dict)


class StrategySignal(StrategyModel):
    id: str
    category: Literal[
        "search_health",
        "compass",
        "compensation",
        "source",
        "artifact",
        "historical",
        "workspace",
    ]
    label: str
    message: str
    basis: str
    severity: Literal["info", "caution", "positive"] = "info"
    confidence: StrategyConfidence
    source_inputs: dict[str, Any] = Field(default_factory=dict)


class StrategyActionCandidate(StrategyModel):
    id: str
    category: Literal[
        "review_workspace_targets",
        "review_compensation_target",
        "review_skill_gap_plan",
        "review_artifact_strategy",
        "review_source_strategy",
        "archive_or_pause_track_review",
        "create_followup_plan_preview",
        "review_search_focus",
    ]
    title: str
    rationale: str
    basis: str
    confidence: StrategyConfidence
    source_inputs: dict[str, Any] = Field(default_factory=dict)
    advisory_only: Literal[True] = True


class StrategySampleSize(StrategyModel):
    opportunities: int = 0
    applications: int = 0
    submitted_applications: int = 0
    responses: int = 0
    compass_evaluations: int = 0
    artifact_performance_records: int = 0


class CompensationAlignmentSummary(StrategyModel):
    summary: str
    basis: str
    confidence: StrategyConfidence
    observations: list[dict[str, Any]] = Field(default_factory=list)


class RoleMarketPositioningSummary(StrategyModel):
    summary: str
    basis: str
    confidence: StrategyConfidence
    themes: list[str] = Field(default_factory=list)


class StrategyRetrospective(StrategyModel):
    summary: str
    basis: str
    confidence: StrategyConfidence
    notes: list[str] = Field(default_factory=list)


class SearchTrackStrategySummary(StrategyModel):
    contract_version: Literal["careero.contracts.v1"] = "careero.contracts.v1"
    workspace_id: uuid.UUID
    workspace_name: str
    generated_at: datetime
    summary: str
    basis: str
    confidence: StrategyConfidence
    sample_size: StrategySampleSize
    source_inputs: dict[str, Any] = Field(default_factory=dict)
    known_uncertainty: list[str] = Field(default_factory=list)
    insufficient_data: list[StrategyInsufficientDataItem] = Field(
        default_factory=list
    )
    signals: list[StrategySignal] = Field(default_factory=list)
    compensation_alignment: CompensationAlignmentSummary
    skill_gap_themes: list[StrategySignal] = Field(default_factory=list)
    role_market_positioning: RoleMarketPositioningSummary
    career_narrative_themes: list[StrategySignal] = Field(default_factory=list)
    retrospective: StrategyRetrospective
    action_candidates: list[StrategyActionCandidate] = Field(default_factory=list)
    warnings: list[str] = Field(default_factory=list)


class CrossTrackStrategyTrack(StrategyModel):
    workspace_id: uuid.UUID
    workspace_name: str
    summary: str
    sample_size: dict[str, Any] = Field(default_factory=dict)
    signal_count: int
    warning_count: int


class CrossTrackStrategyComparison(StrategyModel):
    generated_at: datetime
    basis: str
    confidence: StrategyConfidence
    tracks: list[CrossTrackStrategyTrack] = Field(default_factory=list)
    signals: list[StrategySignal] = Field(default_factory=list)
    insufficient_data: list[StrategyInsufficientDataItem] = Field(
        default_factory=list
    )
    warnings: list[str] = Field(default_factory=list)


class CareerStrategySummary(StrategyModel):
    contract_version: Literal["careero.contracts.v1"] = "careero.contracts.v1"
    generated_at: datetime
    summary: str
    workspace_id: uuid.UUID | None = None
    workspace_name: str | None = None
    active_track: SearchTrackStrategySummary | None = None
    tracks: list[SearchTrackStrategySummary] = Field(default_factory=list)
    cross_track_comparison: CrossTrackStrategyComparison | None = None
    warnings: list[str] = Field(default_factory=list)
